backend/src/apiserver/resource/resource_manager.go

```diff
@@ -887,9 +888,28 @@ func (r *ResourceManager) ReportWorkflowResource(ctx context.Context, workflow *
 	if jobId == "" {
 		// If a run doesn't have job ID, it's a one-time run created by Pipeline API server.
 		// In this case the DB entry should already been created when argo workflow CR is created.
-		err := r.runStore.UpdateRun(runId, condition, workflow.FinishedAt(), workflow.ToStringForStore())
-		if err != nil {
-			return util.Wrap(err, "Failed to update the run.")
+		updateError := r.runStore.UpdateRun(runId, condition, workflow.FinishedAt(), workflow.ToStringForStore())
+		if updateError != nil {
```

Comment:
nit: missed this last time
```suggestion
		if updateError := r.runStore.UpdateRun(runId, condition, workflow.FinishedAt(), workflow.ToStringForStore()); updateError != nil {
```

Reply:
Otherwise LGTM